### PR TITLE
Change the functions for kubetest2-gke deployer to public

### PR DESCRIPTION
**Pull request by chizhg**
Change the functions for kubetest2-gke deployer to public, so that it can be used as a library.

### CODE REVIEW

File: kubetest2-gke/deployer/options/up.go [hunk 1]
```diff
@@ -19,7 +19,28 @@ package options
...
+
+	PrivateClusterAccessLevel    string   `flag:"~private-cluster-access-level" desc:"Private cluster access level, if not empty, must be one of 'no', 'limited' or 'unrestricted'. See the details in https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters."`
+	PrivateClusterMasterIPRanges []string `flag:"~private-cluster-master-ip-range" desc:"Private cluster master IP ranges. It should be IPv4 CIDR(s), and its length must be the same as the number of clusters if private cluster is requested."`
+	SubnetworkRanges             []string `flag:"~subnetwork-ranges" desc:"Subnetwork ranges as required for shared VPC setup as described in https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-shared-vpc#creating_a_network_and_two_subnets. For multi-project profile, it is required and should be in the format of 10.0.4.0/22 10.0.32.0/20 10.4.0.0/14,172.16.4.0/22 172.16.16.0/20 172.16.4.0/22, where the subnetworks configuration for different project are separated by comma, and the ranges of each subnetwork configuration is separated by space."`
```
Comment (amwat) [inline kubetest2-gke/deployer/options/up.go:43]: Thank you for this long overdue refactoring!

**Review by amwat — COMMENTED** [review]
/lgtm

one quick question: there's probably more value in actually exporting the *non-deployer* "helper" functions (i.e. network setup etc.)

Usability of deployers as libraries is questionable considering the contract between deployers and testers is still binary invocation.

File: kubetest2-gke/deployer/options/up.go (hunk 1 above)
Comment (chizhg) [reply kubetest2-gke/deployer/options/up.go:43]: yw!

**Comment by chizhg** [comment]
> /lgtm
> 
> one quick question: there's probably more value in actually exporting the _non-deployer_ "helper" functions (i.e. network setup etc.)
> 
> Usability of deployers as libraries is questionable considering the contract between deployers and testers is still binary invocation.

The major/only reason we want this change is to decouple the different steps (project setup, network setup, cluster creation, etc.) in the pipeline, so that we can run each step individually.

We are not planning to rebuild a new pipeline by calling these public functions, that'll be still orchestrated by the kubetest2 binary.

Does it answer your question?

**Comment by amwat** [comment]
My main question is around how we plan to use the public deployer functions. If users still needs to construct the kubetest2 gke deployer then it's probably less ideal than exporting the networking pieces directly.

But either way, this will help in making these methods reusable by users who want to manage the network lifecycle on their own.

/lgtm
/approve

**Pull request merged**